system/lib/wasmfs/file.h

```diff
@@ -158,6 +158,13 @@ class Directory : public File {
       lockedInserted.setParent(file);
     }
 
+    void unlinkEntry(std::string pathName) {
+      // TODO: Unlinking the parent pointer after erasing from entries.
+      auto unlinked = getDir()->entries[pathName];
+      unlinked->locked().setParent(std::shared_ptr<File>());
+      getDir()->entries.erase(pathName);
```

Comment:
Would it make sense to hold the file lock while the directory entry is cleared as well? That would make the unlink atomic and ensure that no other part of the system can observe the half-unlinked state.

Reply:
Yep, makes sense. Discussed in our meeting